### PR TITLE
Persist import parameters

### PR DESCRIPTION
**Pull request by etiennehomer**

### CODE REVIEW

**Review by dbraquart — CHANGES_REQUESTED** [review]
code: 2 remarks for unused code
tests: ok, but i could not perform the study copy test case (not related to this PR)

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 1]
```diff
@@ -309,6 +311,14 @@ private void duplicateStudyAsync(BasicStudyInfos basicStudyInfos, StudyEntity so
         }
     }
 
+    public static ImportParametersEntity toEntity(ImportParametersInfos importParametersInfos) {
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java:314]: I was used once. Now it's twice with the change of your prevent comment.

File: src/main/java/org/gridsuite/study/server/service/ConsumerService.java [hunk 2]
```diff
@@ -327,7 +333,7 @@ public Consumer<Message<String>> consumeCaseImportSucceeded() {
...
                     DynamicSimulationParametersInfos dynamicSimulationParameters = DynamicSimulationService.getDefaultDynamicSimulationParameters();
                     VoltageInitParametersInfos voltageInitParametersInfos = VoltageInitService.getDefaultVoltageInitParameters();
-                    studyService.insertStudy(studyUuid, userId, networkInfos, caseFormat, caseUuid, caseName, LoadflowService.toEntity(loadFlowParameters, List.of()), ShortCircuitService.toEntity(shortCircuitParameters), DynamicSimulationService.toEntity(dynamicSimulationParameters, objectMapper), VoltageInitService.toEntity(voltageInitParametersInfos), importReportUuid);
+                    studyService.insertStudy(studyUuid, userId, networkInfos, caseFormat, caseUuid, caseName, LoadflowService.toEntity(loadFlowParameters, List.of()), ShortCircuitService.toEntity(shortCircuitParameters), DynamicSimulationService.toEntity(dynamicSimulationParameters, objectMapper), VoltageInitService.toEntity(voltageInitParametersInfos), new ImportParametersEntity(null, importParametersAsStrings), importReportUuid);
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/study/server/service/ConsumerService.java]: I changed a bit the code to be more homogeneous and to use toEtntity()

File: src/test/java/org/gridsuite/study/server/utils/TestUtils.java [hunk 3]
```diff
@@ -80,6 +81,29 @@ public static StudyEntity createDummyStudy(UUID networkUuid, UUID caseUuid, Stri
             .build();
     }
 
+    public static StudyEntity createDummyStudy(UUID networkUuid, UUID caseUuid, ImportParametersEntity importParametersEntity) {
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/test/java/org/gridsuite/study/server/utils/TestUtils.java]: Removed

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 4]
```diff
@@ -309,6 +311,14 @@ private void duplicateStudyAsync(BasicStudyInfos basicStudyInfos, StudyEntity so
...
+        return new ImportParametersEntity(null, importParametersInfos.getParameters());
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java]: Yes done

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 5]
```diff
@@ -309,6 +311,14 @@ private void duplicateStudyAsync(BasicStudyInfos basicStudyInfos, StudyEntity so
...
+    }
+
+    public static ImportParametersInfos fromEntity(ImportParametersEntity importParametersEntity) {
+        return new ImportParametersInfos(importParametersEntity.getParameters());
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java]: Yes done

[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java]: Yes done

**Review by dbraquart — CHANGES_REQUESTED** [review]
code: ok
tests: 1 issue found

- create a study "S" from existing case, with at least one modified import param
- in database, check that this param exists for S (a line in import_parameters_entity_parameters table)
- duplicate "S" into "S (1)"
- in database, check that this param exists for S (1)
- but the param for S has been deleted !

**Comment by etiennehomer** [comment]
> code: ok tests: 1 issue found
> 
> * create a study "S" from existing case, with at least one modified import param
> * in database, check that this param exists for S (a line in import_parameters_entity_parameters table)
> * duplicate "S" into "S (1)"
> * in database, check that this param exists for S (1)
> * but the param for S has been deleted !

Fixed with : 
public static ImportParametersInfos fromEntity(ImportParametersEntity importParametersEntity) {
        return new ImportParametersInfos(importParametersEntity == null ? Map.of() : **Map.copyOf**(importParametersEntity.getParameters()));
    }

Before the fix, I was taking a map that was a JPA entity, using this map in a DTO and then inserting this map in a new ImportParamsEntity. So the data were lost for the source study.

File: src/main/java/org/gridsuite/study/server/service/ConsumerService.java [hunk 6]
```diff
@@ -58,19 +59,25 @@ public class ConsumerService {
     NotificationService notificationService;
     StudyService studyService;
     CaseService caseService;
+    NetworkStoreService networkStoreService;
```
Comment (Meklo) [inline src/main/java/org/gridsuite/study/server/service/ConsumerService.java:62]: It is unused, I reckon you can remove it

Comment (etiennehomer) [reply src/main/java/org/gridsuite/study/server/service/ConsumerService.java]: Yes removed

**Review by dbraquart — APPROVED** [review]
no review/test since last time

File: src/main/java/org/gridsuite/study/server/dto/ImportParametersInfos.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+/**
+ * @author Etienne Homer <etienne.homer at rte-france.com>
+ */
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class ImportParametersInfos {
+    private Map<String, String> parameters = new HashMap<>();
+}
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/dto/ImportParametersInfos.java:26]:
```
public class ImportParametersInfos extends HashMap<String, String> {
    public ImportParametersInfos(Map<String, String> values) {
        super(values);
    }

    public ImportParametersEntity toEntity() {
        return new ImportParametersEntity(null, this);
    }
}
```

File: src/main/java/org/gridsuite/study/server/repository/ImportParametersEntity.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+@Table(name = "importParameters")
+public class ImportParametersEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "id")
+    private UUID id;
+
+    @ElementCollection
+    @CollectionTable
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/repository/ImportParametersEntity.java:31]: To change the table name specify a value or remove this annotation

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/repository/ImportParametersEntity.java [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    private Map<String, String> parameters;
+}
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/repository/ImportParametersEntity.java:33]:
```
    public ImportParametersInfos toDto() {
        return new ImportParametersInfos(parameters);
    }
```

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 10]
```diff
@@ -1249,6 +1249,14 @@ public ResponseEntity<DynamicSimulationParametersInfos> getDynamicSimulationPara
         return ResponseEntity.ok().body(studyService.getDynamicSimulationParameters(studyUuid));
     }
 
+    @GetMapping(value = "/studies/{studyUuid}/import/parameters")
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/study/server/StudyController.java]: Endpoint is removed

File: src/main/java/org/gridsuite/study/server/repository/ImportParametersEntity.java (hunk 8 above)
Comment (etiennehomer) [reply src/main/java/org/gridsuite/study/server/repository/ImportParametersEntity.java:31]: Done

File: src/main/java/org/gridsuite/study/server/dto/ImportParametersInfos.java (hunk 7 above)
Comment (etiennehomer) [reply src/main/java/org/gridsuite/study/server/dto/ImportParametersInfos.java:26]: done

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**